### PR TITLE
Issue #SB-19733 fix: enhancement feature added

### PR DESCRIPTION
**Pull request by raghav14**

### CODE REVIEW

File: src/app/faq-report-issue/faq-report-issue.page.ts [hunk 1]
```diff
@@ -439,17 +448,13 @@ export class FaqReportIssuePage implements OnInit, OnDestroy {
...
-        correlationlist.push({ id: this.formValues.children.subcategory.medium.name, type: CorReleationDataType.MEDIUM }) : undefined;
+      correlationlist.push({ id: this.extractPrepareedFieldStr('medium'), type: CorReleationDataType.MEDIUM });
       // Grade
-      this.formValues.children.subcategory.grade && this.formValues.children.subcategory.grade.name ?
-        correlationlist.push({ id: this.formValues.children.subcategory.grade.name, type: CorReleationDataType.CLASS }) : undefined;
+      correlationlist.push({ id: this.extractPrepareedFieldStr('grade'), type: CorReleationDataType.CLASS });
       // Subject
-      this.formValues.children.subcategory.subject && this.formValues.children.subcategory.subject.name ?
-        correlationlist.push({ id: this.formValues.children.subcategory.subject.name, type: CorReleationDataType.SUBJECT }) : undefined;
+      correlationlist.push({ id: this.extractPrepareedFieldStr('subject'), type: CorReleationDataType.SUBJECT });
```
Comment (swayangjit) [inline src/app/faq-report-issue/faq-report-issue.page.ts:457]: What will be the cdata.id if  user is selected more than one subject ?

File: src/app/faq-report-issue/faq-report-issue.page.ts [hunk 2]
```diff
@@ -350,23 +350,32 @@ export class FaqReportIssuePage implements OnInit, OnDestroy {
     }
   }
 
+  extractPrepareedFieldStr(field) {
```
Comment (swayangjit) [inline src/app/faq-report-issue/faq-report-issue.page.ts:353]: Typo 'extractPrepareedFieldStr'

File: src/app/faq-report-issue/faq-report-issue.page.ts (hunk 1 above)
Comment (raghav14) [reply src/app/faq-report-issue/faq-report-issue.page.ts:457]: Same "Subject"

Comment (swayangjit) [reply src/app/faq-report-issue/faq-report-issue.page.ts:457]: it will be an Array or String ?

**Pull request merged**